src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/RuntimeFunctionsTableNode.cs

```diff
@@ -72,31 +71,86 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
+            List<uint> mapping = new List<uint>();
 
-                for (int frameIndex = 0; frameIndex < method.FrameInfos.Length; frameIndex++)
+            for (int hot = 0; hot < 2; hot++)
+            {
+                foreach (MethodWithGCInfo method in _methodNodes)
                 {
-                    FrameInfo frameInfo = method.FrameInfos[frameIndex];
+                    int[] funcletOffsets = method.GCInfoNode.CalculateFuncletOffsets(factory);
+                    int startIndex;
+                    int endIndex;
 
-                    // StartOffset of the runtime function
-                    int codeDelta = 0;
-                    if (Target.Architecture == TargetArchitecture.ARM)
+                    if (hot == 0)
+                    {
+                        startIndex = 0;
+                        endIndex = method.FrameInfos.Length;
+                    }
+                    else if (method.GetColdCodeNode() == null)
+                    {
+                        continue;
+                    }
+                    else
                     {
-                        // THUMB_CODE
-                        codeDelta = 1;
+                        Debug.Assert((method.FrameInfos.Length + method.ColdFrameInfos.Length) == funcletOffsets.Length);
+                        startIndex = method.FrameInfos.Length;
+                        endIndex = funcletOffsets.Length;
                     }
-                    runtimeFunctionsBuilder.EmitReloc(method, RelocType.IMAGE_REL_BASED_ADDR32NB, delta: frameInfo.StartOffset + codeDelta);
-                    if (!relocsOnly && Target.Architecture == TargetArchitecture.X64)
+
+                    for (int frameIndex = startIndex; frameIndex < endIndex; frameIndex++)
                     {
-                        // On Amd64, the 2nd word contains the EndOffset of the runtime function
-                        runtimeFunctionsBuilder.EmitReloc(method, RelocType.IMAGE_REL_BASED_ADDR32NB, delta: frameInfo.EndOffset);
+                        FrameInfo frameInfo;
+                        ISymbolNode symbol;
+
+                        if (frameIndex >= method.FrameInfos.Length)
+                        {
+                            frameInfo = method.ColdFrameInfos[frameIndex - method.FrameInfos.Length];
+                            symbol = method.GetColdCodeNode();
+
+                            if (frameIndex == method.FrameInfos.Length)
+                            {
+                                mapping.Add(runtimeFunctionIndex);
+                                mapping.Add((uint)_insertedMethodNodes[method]);
+                            }
+                        }
+                        else
+                        {
+                            frameInfo = method.FrameInfos[frameIndex];
+                            symbol = method;
+                        }
+
+                        // StartOffset of the runtime function
+                        int codeDelta = 0;
```

Comment:
One more place that would benefit from some form of unification of the <code>ArmCodeDelta</code> calculation - apparently the idiom appears in multiple functions so I think it would be cleanest to make it a property of <code>TargetDetails</code>.